gc/base/GCExtensionsBase.hpp

```diff
@@ -395,6 +395,8 @@ class MM_GCExtensionsBase : public MM_BaseVirtual {
 	bool gcThreadCountForced; /**< true if number of GC threads is specified in java options. Currently we have a few ways to do this:
 										-Xgcthreads		-Xthreads= (RT only)	-XthreadCount= */
 
+	bool snapshotAtTheBeginningBarrierEnabled;
```

Comment:
Do we need this boolean?  Can we just use the write barrier type?

Reply:
For now we want to use Xgc option to specify when to use sanpshot at the beginning. writeBarrierType is a field in Configuration and it seems that configuration is initialized after Xgc options are parsed. I added a check for snapshotAtTheBeginningBarrierEnabled in MM_ConfigurationDelegate to set the barrier type.. Is there a better way?

Reply:
I think @ebadyano is right that we do need something in Extensions to at least temporary store the result  of parsing command line options, till the writeBarrierType is initialized in the Delegate.

I can only suggest minor modifications:
1) move the field to extensions->configurationOptions struct (where there are a few other high level  params controlled by command line options). Try make the name consistant with others in the struct,  for example: _forceOptionWriteBarrierSATB.
2) do not use that field anywhere else in the code, but fetch the barrier type from the Delegate instead.

Reply:
added _forceOptionWriteBarrierSATB under extensions->configurationOptions.